### ISSUE TITLE
Double quoted font-names break style attribute when the target element already has an inline style

### ISSUE THREAD

**Issue by tborg** [Stranger6667/css-inline#392]
Hi! Thank you for this excellent library.

I wanted to revisit #333 — this problem arose for us under a very specific circumstance: when the font family is inlined into an element that already has an inline style. Here is a minimal repro that will demonstrate the problem in the playground:

```
<html>
  <head>
    <style>.text { font-family: "Segoe UI"; }</style>
  </head>
  <body>
    <span class="text" style="whitespace:nowrap">Big Text</span>
  </body>
</html>
```

results in output like this:

```
<html><head>
    
  </head>
  <body>
    <span class="text" style="font-family: "Segoe UI";whitespace: nowrap">Big Text</span>
  
</body></html>
```

Maybe the reason why will be immediately obvious to you — otherwise, if you can give me a pointer on where to get started I could take a crack at it.

Thanks!

**Comment by Stranger6667** [comment]
Hi! Thank you for your kind words!

I can reproduce the issue with the latest version (with #333 I wasn't able to do it). The logic for double quotes handling is [here](https://github.com/Stranger6667/css-inline/blob/master/css-inline/src/html/serializer.rs#L403-L424):

```rust
    if value.as_bytes().contains(&b'"') {
        // Roughly based on `str::replace`
        let mut last_end = 0;
        for (start, part) in value.match_indices('"') {
            writer.write_all(
                value
                    .get(last_end..start)
                    .expect("Invalid substring")
                    .as_bytes(),
            )?;
            writer.write_all(b"'")?;
            last_end = start.checked_add(part.len()).expect("Size overflow");
        }
        writer.write_all(
            value
                .get(last_end..value.len())
                .expect("Invalid substring")
                .as_bytes(),
        )?;
    } else {
        writer.write_all(value.as_bytes())?;
    };
```

Right now I am busy with other things, but I will have the capacity to take a look at it somewhere next week. However, I am happy to review a PR :)

**Comment by Stranger6667** [comment]
Apparently, it is not used in all cases during [styles merge](https://github.com/Stranger6667/css-inline/blob/master/css-inline/src/html/serializer.rs#L455).

**Comment by Stranger6667** [comment]
[This macro](https://github.com/Stranger6667/css-inline/blob/master/css-inline/src/html/serializer.rs#L428) does not use the proper version of quote handling and should be like this:


```rust
macro_rules! push_or_update {
    ($style_buffer:expr, $length:expr, $name: expr, $value:expr) => {{
        if let Some(style) = $style_buffer.get_mut($length) {
            style.clear();
            write_declaration(style, $name, $value)?;
        } else {
            let value = $value.trim();
            let mut style = Vec::with_capacity(
                $name
                    .len()
                    .saturating_add(STYLE_SEPARATOR.len())
                    .saturating_add(value.len()),
            );
            write_declaration(&mut style, $name, $value)?;
            $style_buffer.push(style);
        };
        $length = $length.saturating_add(1);
    }};
}
```

Then this test passes:


```rust
#[test]
fn font_already_with_quoted() {
    assert_inlined!(
        style = r#".text { font-family: "Segoe UI"; }"#,
        body = r#"<h1 class="text" style="whitespace:nowrap">Hello world!</h1>"#,
        expected = r#"<h1 class="text" style="font-family: 'Segoe UI';whitespace: nowrap">Hello world!</h1>"#
    )
}

```

**Comment by Stranger6667** [comment]
Feel free to create a PR with changes like above (and changelog entries for both, the rust crate & python bindings) :)

**Comment by tborg** [comment]
Awesome — thanks for the guidance.

**Comment by Stranger6667** [comment]
Released in `0.14.2`
